### DISCUSSION TITLE
building ARM64 code with udk 2018 and VS 2017

### DISCUSSION THREAD

**Discussion by sergiotarchi** [tianocore/edk2#3681] [Q&A]
Hi,
I setup my environment with udk 2018 and VS 2017.
I can see in the tools_def.txt that it's not possible to define a tool chain, for udk 2018, to build ARM64 (AARCH64) code with VS 2017, it only allows ARM.
I have seen that with some simple change on the tools_def.txt file I can enable a tool chain for AARCH64 code on VS 2017 but I would like to ask if there's some known issue in doing this, or this should work.
Thanks.

**Comment by spbrogan** [comment]
any reason you are using UDK 2018.  That is 4 + years old.  
If you look at the stable tags on "master" you will see they have been created every 3 months for the last few years and in those newer releases you will have a much better chance of building with aarch64.  

There are numerous limitations even in the latest code for building aarch64 with visual studio.  Many of the needed assembly files don't exist.  This means it is possible to build some packages and some libraries, but you can't build the full tree or any given aarch64 platform (ArmVirtPkg is a Aarch64 qemu platform).   So if you just want to compile a UEFI shell application you can probably do that but a full open source platform isn't available (that i am aware of).

**Reply by sergiotarchi** [reply]
Hi.
I'm using udk 2018 because my purpose is to build an efi application for arm64 (efi boot kernel), porting with minimum changes my application originally written with udk 2010 to run on x64.
Udk 2018 supports visual studio 2017 and visual studio 2017 supports arm64 build.
I also tried with a newer edk2 version but it seems that more and more changes are needed on the original code.
For example, as first, I can't find the module StdLib in edk2 2022, and it's among the pacakges used by the application I want to build. Do you have some suggestion on this?
Thanks.
